agent/stats/service_connect_linux.go

```diff
@@ -117,7 +117,17 @@ func convertToTACSStats(mf map[string]*prometheus.MetricFamily, taskId string) (
 					}
 					metricCounts = append(metricCounts, &metricCount)
 				}
-				metricCounts = convertHistogramMetricCounts(metricCounts)
+
+				metricValues, metricCounts = convertHistogramMetricCounts(metricValues, metricCounts)
+
+				// If all values are 0 in metricCount, then no need to send the metrics to TACS
+				if len(metricCounts) == 0 {
```

Comment:
As an optimization we can pre-compute this by checking the SampleCount value to be 0 (`metric.Histogram.SampleCount` I think) on L99 and completely skip processing these buckets?

Reply:
sure. I would take this up as a follow-up task and address that in a separate PR.